src/olympia/addons/models.py

```diff
@@ -487,12 +487,15 @@ def delete(self, msg='', reason=''):
     @classmethod
     def initialize_addon_from_upload(cls, data, is_listed=True):
         fields = cls._meta.get_all_field_names()
-        # Reclaim GUID from deleted add-on.
-        try:
-            old_guid_addon = Addon.unfiltered.get(guid=data['guid'])
-            old_guid_addon.update(guid=None)
-        except ObjectDoesNotExist:
-            old_guid_addon = None
+        guid = data.get('guid')
```

Comment:
Since `guid` can't be `""` (empty string) I'm a fan of explicitly doing `data.get('guid', None)` here and check for `if guid is not None`